autograder-core/src/main/java/de/firemage/autograder/core/ProblemType.java

```diff
@@ -1,16 +1,63 @@
 package de.firemage.autograder.core;
 
 public enum ProblemType {
+    /**
+     * If the code is split into multiple packages, all input must happen in one package. Otherwise, one class must do all input.
+     * <br/>
+     * Has false positives.
+     */
+    @HasFalsePositives
     UI_INPUT_SEPARATION,
+
+    /**
+     * If the code is split into multiple packages, all output must happen in one package. Otherwise, one class must do all output.
+     * <br/>
+     * Has false positives.
+     */
+    @HasFalsePositives
     UI_OUTPUT_SEPARATION,
+
+    /**
+     * Checks for non-private inner types (static classes, interfaces, enums, local types)
+     */
```

Comment:
```suggestion
    /**
     * Checks for non-private inner types (static classes, interfaces, enums, local types)
     * <br>
     * This check ignores inner classes that are not static, because there are cases where
     * that can be useful.
     */
```